// errorHandler.js

const handleAsync = (fn) => (req, res, next) => {
    Promise.resolve(fn(req, res, next)).catch(next);
  };
  
  const notFound = (req, res, next) => {
    const error = new Error(`Not Found - ${req.originalUrl}`);
    res.status(404);
    next(error);
  };
  
  const errorHandler = (err, req, res, next) => {
    const statusCode = res.statusCode === 200 ? 500 : res.statusCode;
    res.status(statusCode);
    res.json({
      error: err.message,
      stack: process.env.NODE_ENV === 'production' ? 'ðŸ¥ž' : err.stack,
    });
  };
  
  module.exports = { handleAsync, notFound, errorHandler };
  